src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/VisualStudioNuGetProject.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.ProjectManagement;
+using NuGet.Protocol.Core.Types;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public class VisualStudioNuGetProject : NuGetProject
+    {
+        private const string LiveShareUriScheme = "vsls";
+        private const string ProjectGuidQueryString = "projectGuid";
+        private readonly string _projectGuidString;
+
+        public VisualStudioNuGetProject(Guid projectGuid)
```

Comment:
This name is a bit confusing to me, given that there's things like https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsMSBuildNuGetProject.cs and https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectJsonNuGetProject.cs. 

This is just a wrapped around it all. 

Given that it's only used in the codespaces case, canwe move the type to that assembly instead of the VS generic one?

Reply:
The name was suggested by @zivkan.

I can move it to NuGet.VisualStudio.Internal.Contracts but I cannot move it to OnlineEnvironment.Client, would you rather see it in the contracts assembly?